sidestep_engine/gui/server.py

```
@@ -438,15 +438,20 @@ async def list_models(checkpoint_dir: str = ""):
         from sidestep_engine.gui.file_ops import scan_models
         return JSONResponse(scan_models(checkpoint_dir))
 
+    # NOTE: the dataset/audio scans below walk every file in the configured
+    # roots (duration probe per audio file).  They MUST run off the event
+    # loop via asyncio.to_thread — inline they froze the whole GUI for the
+    # length of a full library scan (~7 s for 2.5k files).
+
     @app.get("/api/datasets")
     async def list_datasets(tensors_dir: str = ""):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `list_datasets`

(ANN202)